Ternary Search Tree/TernarySearchTree.swift

```diff
@@ -33,52 +33,52 @@ public class TernarySearchTree<Element> {
 
     - returns: Value indicating insertion success/failure.
     */
-    public func insert(data: Element, withKey key: String) -> Bool {
-        return insertNode(&root, withData: data, andKey: key, atIndex: 0)
+    @discardableResult public func insert(data: Element, withKey key: String) -> Bool {
+        return insert(node: &root, withData: data, andKey: key, atIndex: 0)
     }
 
     /**
      Helper method for insertion that does the actual legwork. Insertion is performed recursively.
 
-     - parameter aNode:     The current node to insert below.
+     - parameter node:     The current node to insert below.
      - parameter data:      The data being inserted.
      - parameter key:       The key being used to find an insertion location for the given data
      - parameter charIndex: The index of the character in the key string to use to for the next node.
 
      - returns: Value indicating insertion success/failure.
      */
-    private func insertNode(inout aNode: TSTNode<Element>?, withData data: Element, andKey key: String, atIndex charIndex: Int) -> Bool {
+    private func insert(node: inout TSTNode<Element>?, withData data: Element, andKey key: String, atIndex charIndex: Int) -> Bool {
 
         //sanity check.
         if key.characters.count == 0 {
             return false
         }
 
         //create a new node if necessary.
-        if aNode == nil {
-            let index = key.startIndex.advancedBy(charIndex)
-            aNode = TSTNode<Element>(key: key[index])
+        if node == nil {
+            let index = key.index(key.startIndex, offsetBy: charIndex)
+            node = TSTNode<Element>(key: key[index])
         }
 
```

Comment:
You can probably add a guard statement here, and return false for whatever reason the node failed to create.